### ISSUE TITLE
Better backport commands

### ISSUE THREAD

**Issue by krassowski** [scientific-python/MeeseeksDev#115]
I wonder if backporting instructions would be more helpful if instead of:

```bash
git checkout 3.6.x
git pull
git cherry-pick -x -m1 1728207ffbad5baea1e77e4005230632bf6f16a6
git commit -am 'Backport PR #15279: Update benchmark reporter snapshots'
git push YOURFORK 3.6.x:auto-backport-of-pr-15279-on-3.6.x
```

it was:

```bash
git checkout 3.6.x
# (change 1)
git pull  # if on a fork, use `git fetch upstream && git merge upstream/3.6.x`
# (change 2)
git checkout -b backport-of-pr-15279-on-3.6.x 
git cherry-pick -x -m1 1728207ffbad5baea1e77e4005230632bf6f16a6
git commit -am 'Backport PR #15279: Update benchmark reporter snapshots'
# (change 3)
git push --set-upstream origin backport-of-pr-15279-on-3.6.x
```

Why?

1. Larger projects often work using forks where `git pull` is often not enough
2. Creating a new branch early avoids having to reset our old branch when doing multiple backports
3. Is not strictly required for users with `push.autoSetupRemote` but I personally find it safer to have an explicit target

What do you all think?

**Comment by Carreau** [comment]
yeah, we can update if you wish, in general we may want also reset --hard
upstream/tag, but as many instructions those will often vary, for example
in your instructions you refer to upstream with the upstream name, but it
often might not be the case.

I believe there is even a way to create a branch based on a tag in a single
command

git checkout -b autobacport-xxx origin/3.6.x

**Comment by flying-sheep** [comment]
I had the exact same thoughts and made #168 (only difference is that I didn’t think of `--set-upstream`).

So I think these instructions are probably an almost ideal shape as multiple people came up with the same ones.

In my version, I address the `upstream` and `origin` thing by mentioning them, clarifying where each branch is supposed to come from / go to.
